homeassistant/components/sensor/hp_ilo.py

```diff
@@ -84,15 +96,30 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 class HpIloSensor(Entity):
     """Representation of a HP ILO sensor."""
 
-    def __init__(self, hp_ilo_data, sensor_type, client_name):
+    def __init__(self, hass, hp_ilo_data, sensor_type, sensor_name,
+                 sensor_value_template, unit_of_measurement):
         """Initialize the sensor."""
-        self._name = '{} {}'.format(client_name, sensor_type[0])
-        self._ilo_function = sensor_type[1]
-        self.client_name = client_name
+        self._hass = hass
+        self._name = sensor_name
+        self._unit_of_measurement = unit_of_measurement
+        self._ilo_function = SENSOR_TYPES[sensor_type][1]
         self.hp_ilo_data = hp_ilo_data
 
+        if sensor_value_template is not None:
+            # Make sure the template always returns valid json
```

Comment:
We should not use Jinja2 templates to generate JSON that we then will load again. Jinja2 should be used to just generate the value.

Example template could be `{{ value.something[0].name }}`